fulp_modules/main_features/bloodsuckers/code/bloodsucker/bloodsucker_life.dm

```diff
@@ -121,17 +121,19 @@
 		if(amInCoffin && HAS_TRAIT(C, TRAIT_NODEATH))
 			if(poweron_masquerade)
 				to_chat(C, "<span class='warning'>You will not heal while your Masquerade ability is active.</span>")
+				return
 			fireheal = min(C.getFireLoss_nonProsthetic(), actual_regen)
 			mult *= 5 // Increase multiplier if we're sleeping in a coffin.
 			costMult /= 2 // Decrease cost if we're sleeping in a coffin.
 			C.extinguish_mob()
 			C.remove_all_embedded_objects() // Remove Embedded!
 			if(check_limbs(costMult))
 				return TRUE
-		/// In Torpor, but not in a Coffin? Heal faster anyways.
+		// In Torpor, but not in a Coffin? Heal Burn and slightly faster.
```

Comment:
Check_End_Torpor still ends torpor immediately if you're not in a coffin, and its call is based on total_brute still.  This will cause you to enter and begin torpor multiple times in quick succession

Now would be a good time to just handle all the checks in Check_End_Torpor and Check_End_Torpor